test/visitors/test_bpmn_connectivity_visitor.py

```diff
@@ -0,0 +1,43 @@
+from bpmncwpverify.visitors.bpmn_connectivity_visitor import BpmnConnectivityVisitor
+import pytest
+from bpmncwpverify.core.bpmn import Node
+
+
+def test_ensure_in_and_out_messages(mocker):
+    def setup_node(id, in_msgs=None, out_msgs=None, message_event_definition=""):
+        node = mocker.Mock(spec=Node)
+        node.id = id
+        node.in_msgs = in_msgs or []
+        node.out_msgs = out_msgs or []
+        node.message_event_definition = message_event_definition
+        return node
+
+    visitor = BpmnConnectivityVisitor()
+
+    # Test ensure_in_messages - no message event definition
+    test_node_in = setup_node("123", in_msgs=[1])
+    with pytest.raises(
+        Exception,
+        match="Exception occurred while visiting event:123. A message flow can only go to a Message start or intermediate event; Receive, User, or Service task; Subprocess; or black box pool.",
```

Comment:
Should these be written over a `Result` type that wraps an `Error`? We want the make error handling uniform which means creating `Error` types for things related to BPMN semantic rules.